networkx/drawing/tests/test_pydot.py

```diff
@@ -111,48 +111,46 @@ def test_pydot_issue_7581(tmp_path):
 
 def test_pydot_issue_258():
     G = nx.Graph([("Example:A", 1)])
-    with pytest.raises(ValueError):
-        nx.nx_pydot.to_pydot(G)
-    with pytest.raises(ValueError):
-        nx.nx_pydot.pydot_layout(G)
+    pydot_G = nx.nx_pydot.to_pydot(G)
+    # Confirm that nodes exist in a
+    # Keys as integers aren't preserved in the conversion. They are read as strings.
+    assert {str(n) for n in G.nodes}.issubset(nx.nx_pydot.from_pydot(pydot_G).nodes)
 
     G = nx.Graph()
     G.add_node("1.2", style="filled", fillcolor="red:yellow")
-    with pytest.raises(ValueError):
-        nx.nx_pydot.to_pydot(G)
-    G.remove_node("1.2")
+    assert (
+        G.nodes.data() == nx.nx_pydot.from_pydot(nx.nx_pydot.to_pydot(G)).nodes.data()
+    )
+    G = nx.Graph()
```

Comment:
Does this create duplicate tests? Maybe I'm reading the diff wrong, but it looks like it creates the same test twice (lines 119-124 and then 125-128).  This repeating pattern appears again below for lines 132, 153.